cpp/dolfinx/fem/discreteoperators.h

```diff
@@ -22,6 +22,277 @@
 namespace dolfinx::fem
 {
 
+/// @brief Assemble a discrete curl operator.
+///
+/// Curl of Nedelec(k) function -> RT(k-1) function.
+///
+/// @param[in] V0
+/// @param[in] V1
+/// @param[in] mat_set A functor that sets values in a matrix
```

Comment:
If the curl-operator is specific to the function space, do we need to send in the spaces?
Would `degree-k` be sufficient, or would the dofmap potentially be different due to the graph re-ordering (GPS)?

If we send in the spaces, what kind of checks do we need?

Reply:
This PR isn't ready for review (hence it's a draft). PR was created to provide a cross-reference for when the discrete curl is ready.

Reply:
Ok.